packages/provider-utils/src/secure-json-parse.ts

```diff
@@ -0,0 +1,90 @@
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+const suspectProtoRx = /"__proto__"\s*:/;
+const suspectConstructorRx = /"constructor"\s*:/;
+
+function _parse(text: string) {
+  // utf8 BOM checker (https://github.com/fastify/secure-json-parse/pull/5)
+  if (text && text.charCodeAt(0) === 0xfeff) {
+    text = text.slice(1);
+  }
+
+  // Parse normally
+  const obj = JSON.parse(text);
+
+  // Ignore null and non-objects
+  if (obj === null || typeof obj !== 'object') {
+    return obj;
+  }
+
+  if (
+    suspectProtoRx.test(text) === false &&
+    suspectConstructorRx.test(text) === false
+  ) {
+    return obj;
+  }
+
+  // Scan result for proto keys
+  return filter(obj);
+}
+
+function filter(obj: any) {
+  let next = [obj];
+
+  while (next.length) {
+    const nodes = next;
+    next = [];
+
+    for (const node of nodes) {
+      if (Object.prototype.hasOwnProperty.call(node, '__proto__')) {
+        throw new SyntaxError('Object contains forbidden prototype property');
+      }
+
+      if (
+        Object.prototype.hasOwnProperty.call(node, 'constructor') &&
+        Object.prototype.hasOwnProperty.call(node.constructor, 'prototype')
+      ) {
+        throw new SyntaxError('Object contains forbidden prototype property');
+      }
+
+      for (const key in node) {
+        const value = node[key];
+        if (value && typeof value === 'object') {
+          next.push(value);
+        }
+      }
+    }
+  }
+  return obj;
+}
+
+export function secureJsonParse(text: string) {
+  const { stackTraceLimit } = Error;
+  Error.stackTraceLimit = 0;
+  try {
+    return _parse(text);
+  } finally {
+    Error.stackTraceLimit = stackTraceLimit;
+  }
```

Comment:
@gr2m do you know what this is for? I usually don't like messing w/ globals - but this might be for security reasons?

Reply:
It's for performance reasons, see https://github.com/fastify/secure-json-parse/pull/90. Let me know if you want to keep it or not, I will add a comment with the pull request URL for future reference.

Fun fact: I know @Uzlopak, the author of that PR quite well, he is helping co-maintain @probot and @octokit for years now.

Reply:
let's keep it - might be worth adding a comment that this is for performance reasons

Reply:
ah nevermind the comment is there